src/EnergyPlus/Construction.cc

```diff
@@ -2063,4 +2062,113 @@ void ConstructionProps::setArraysBasedOnMaxSolidWinLayers(EnergyPlusData &state)
     }
 }
 
+void ConstructionProps::writeCacheData(EnergyPlusData &state)
+{
+    // empty cache for this construction
+    nlohmann::json cacheData;
+
+    // construction name
+    cacheData["name"] = this->Name;
+
+    // CTF data
+    Cache::arrayToJSON(this->CTFCross, cacheData, "ctf_cross");
+    Cache::arrayToJSON(this->CTFFlux, cacheData, "ctf_flux");
+    Cache::arrayToJSON(this->CTFInside, cacheData, "ctf_inside");
+    Cache::arrayToJSON(this->CTFOutside, cacheData, "ctf_outside");
+
+    // other necessary data
+    cacheData["num_histories"] = this->NumHistories;
+    cacheData["num_ctf_terms"] = this->NumCTFTerms;
+    cacheData["ctf_timestep"] = this->CTFTimeStep;
+    cacheData["u_value"] = this->UValue;
+
+    // long verification key
+    cacheData["full_data_key"] = this->getCacheKeyString(state);
+
+    // write to cache
+    std::string key = this->getCacheKey(state);
+    state.dataCache->cache[Cache::CTFKey][key] = cacheData;
+}
+
+void ConstructionProps::loadFromCache(EnergyPlusData &state)
```

Comment:
New construction member method for loading the CTF and other cached construction-specific data from the cache.